module3/lessons/refactoring_api_consumption.md

```diff
@@ -0,0 +1,263 @@
+---
+layout: page
+title: Refactoring Patterns for API Consumption
+---
+
+## Setup
+
+We will start with the `consuming-apis-complete` branch of the [Set List API](https://github.com/turingschool-examples/set-list-api/tree/consuming-apis-complete) for this lesson. 
+
+Instructions for using Rails Encrypted Credentials can be found in the "Required Setup" section [here](./testing_tools_for_api_consumption).
+
+## Learning Goals
+
+- Identify SRP and MVC violations with implementing API consumption entirely in the controller
+
+- Refactor code to pull logic out of the controller
+
+- Identify the purpose of a Gateway/Service and a PORO
+
+## Vocabulary
+
+**Refactor** - to change code in a way that the end functionality still works as intended, but reorganizes it in a way to make it easier to maintain, easier to test, etc.
+
+**SRP** - Single Responsibility Principle; the ideal that a piece of code should be responsible for one kind of task (this can be at a class level, a method level, etc.).
+
+**Design Pattern** - an implementation of code which follows as much "industry standard" as possible to achieve clean organization of our code.
+
+**MVC** - "Model, View, Controller" design pattern; a way of organizing our code into logical portions where our "business logic" is managed by the Controller, the "data logic" is managed by the Models, and the "presentation logic" is managed by the Views.
+
+
+## Reminders
+
+1. **There is no one right way** to refactor and structure API consumption in Rails or in any framework. We will introduce you to a few ideas and design patterns but we are not picky about which patterns you use. 
+
+2. Before refactoring anything, it is **essential** that robust testing is already in place. See section on "Red, Green, Refactor Below"
+
+---
+
+## Identifying Candidates for Refactoring
+
+### Warm-Up
+
+1. What anti-patterns do you see in the current implementation of API consumption in the `consuming-apis-complete` branch?
```

Comment:
Do you think it would be helpful to direct them to the Controller file specifically for this exercise, so they're not looking all over the app for these anti-patterns?